tests/request/payload.spec.js

```diff
@@ -1,24 +1,86 @@
+  const expectedBase64JsonPayload = 'eyJhIjoiMSJ9';
+
+  describe('when content type not supported', () => {
+    it('should throw an error', () => {
+      expect(() => new Payload({}, 'random')).toThrow('Payload content type must be specified and one of [application/json,multipart/form-data]');
+    });
+  });
+
+  describe('when content type is json', () => {
+    const json = { a: '1' };
+    const payload = new Payload(json);
+
+    describe('#getContentType', () => {
+      it('should return the payload content type', () => {
+        const payloadJSON = payload.getContentType();
+        expect(payloadJSON).toBe('application/json');
+      });
+    });
+
+    describe('#getPayloadData', () => {
+      it('should return the payload data as a string', () => {
+        const payloadData = payload.getPayloadData();
+        expect(payloadData).toEqual(JSON.stringify(json));
+      });
+    });
+
+    describe('#getBase64Payload', () => {
+      it('should return the payload data as base64 string', () => {
+        const base64Payload = payload.getBase64Payload();
+        expect(base64Payload).toBe(expectedBase64JsonPayload);
+      });
     });
   });
 
-  describe('#getBase64Payload', () => {
-    it('should return the base64Payload string', () => {
-      const base64Payload = amlPayload.getBase64Payload();
-      expect(base64Payload).toBe(expectedBase64Payload);
+  describe('when content type is data form', () => {
+    const dataFormFields = {
+      getFormDataFields: () => [{
+        name: 'one',
+        value: 'two',
+      }],
+    };
+
+    const payload = new Payload(dataFormFields, ContentType.FORM_DATA);
+
+    describe('#getContentType', () => {
+      it('should return the payload content type', () => {
+        const payloadJSON = payload.getContentType();
+        expect(payloadJSON).toBe('multipart/form-data');
+      });
+    });
+
+    describe('#getPayloadData', () => {
+      it('should return the payload data as a buffer', () => {
+        const payloadData = payload.getPayloadData();
+
+        const formData = new FormData();
+        formData.setBoundary(payload.getRawData().getBoundary());
+        const { name, value } = dataFormFields.getFormDataFields()[0];
+        formData.append(name, value);
+
+        const expectedBase64 = formData.getBuffer().toString('base64');
+        expect(payloadData.toString('base64')).toBe(expectedBase64);
+      });
+    });
+
+    describe('#getBase64Payload', () => {
+      it('should return the payload data as base64 string', () => {
+        const base64Payload = payload.getBase64Payload();
+
+        const formData = new FormData();
+        formData.setBoundary(payload.getRawData().getBoundary());
+        const { name, value } = dataFormFields.getFormDataFields()[0];
+        formData.append(name, value);
+
+        const expectedBase64 = formData.getBuffer().toString('base64');
+
+        expect(base64Payload).toBe(expectedBase64);
+      });
```

Comment:
Not sure about these tests.
These are not easy, but here, you are way a bit repeating the implementation...

We could mock `FormData` so we can control what it returns when calling its constructor, so we can make sure that we control what `.getBuffer()` returns, we can also assert that a new FormData is created, and asserting that its `.append()` method is called with the right data ?

Reply:
Yes, but I was thinking of making the test like the ones for json, to have a constant value and tests if the output is the same.
The only issue I had was that the boundary is for the form data is dynamic, so with the new changes I made it static.